sass2scss.cpp

```diff
@@ -788,14 +790,74 @@ namespace Sass
 		}
 	}
 
+	// Preprocess the sass text in order to detect
+	// unfinished lines and, thus, concatenate them
+	// with the following line.
+	std::string preprocess(const std::string& sass)
+	{
+		std::string result = "";
+
+		std::istringstream lines(sass);
+		std::string line;
+
+		bool concat = false;
+		size_t par_count = 0;
+
+		while (safeGetline(lines, line) && !lines.eof())
+		{
+			if (std::regex_match(line, std::regex("[[:space:]]*")))
```

Comment:
I don't know how much overhead the regex object brings, but it does make sense, considering you may have a SASS file with thousands of lines in it.